### DISCUSSION TITLE
Email verification links return 403

### DISCUSSION THREAD

**Discussion by LeandroBerlin** [filamentphp/filament#12693] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2.70

### How can we help you?

I am registering a new user and I receive the email to verify the newly created account but the link returns a 403.

The User model has `MustVerifyEmail` and 

```
public function canAccessPanel(Panel $panel): bool
    {
        return $this->hasVerifiedEmail();
    }
```

What am I missing? Has anyone had this before and knows how to solve it?

**Comment by leandrocfe** [comment, marked as answer]
you shouldn't return $this->hasVerifiedEmail(), because the user needs to access the painel to complete the process

**Reply by tiagoabaptista** [reply]
I've found this as a clearer explanation:

> This is where the FilamentPHP [documentation](https://filamentphp.com/docs/3.x/panels/installation#allowing-users-to-access-a-panel) is slightly misleading. Documentation says that you need to add $this->hasVerifiedEmail() in the canAccessPanel method. But if you're implementing Illuminate\Contracts\Auth\MustVerifyEmail then you don't need that. Instead, it should simply return true

```php
<?php

use Filament\Models\Contracts\FilamentUser;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Filament\Panel;

class User extends Authenticatable implements FilamentUser, MustVerifyEmail
{
    // Your User Model code

    public function canAccessPanel(Panel $panel): bool
    {

        return true;

    }

}
```

https://riz.codes/fix-email-verification-in-filamentphp/
